src/redux/actions/loadUser.ts

```diff
@@ -2,11 +2,32 @@ import {Dispatch} from "react";
 import AsyncStorage from "@react-native-async-storage/async-storage";
 import axios, {AxiosError} from "axios";
 import {URI} from "../../URI";
+import {userLoadFailed, userLoadRequest, userLoadSuccess} from "../types/types";
 
-export const loadUser = () => async (dispatch: Dispatch<any>) => {
+interface LoadUserRequestAction {
+    type: 'userLoadRequest';
+}
+
+interface LoadUserSuccessAction {
+    type: 'userLoadSuccess';
+    payload: { user: string, token: string | null };
+}
+
+interface LoadUserFailedAction {
+    type: 'userLoadFailed';
+    payload: string;
+}
+
+type LoadUserAction =
+    | LoadUserRequestAction
+    | LoadUserSuccessAction
+    | LoadUserFailedAction;
+
+
+export const loadUser = () => async (dispatch: Dispatch<LoadUserAction>) => {
     try {
         dispatch({
```

Comment:
Imagine later we'll want to change the type of action or add a payload to it. We'll have to go to every dispatch and update object literal passed to it. Let's use https://redux-toolkit.js.org/api/createAction. With action creators we'll be able to get rid of interfaces for every action like `LoadUserFailedAction`